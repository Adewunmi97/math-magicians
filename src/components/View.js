/* eslint-disable react/prefer-stateless-function */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from '../assets/view.module.css';

class View extends Component {
  render() {
    const { output, operation, next } = this.props;
    return (
      <div data-testid="screen" className={`${styles.View}`}>
        <p>{output || null}</p>
        <p>{operation || null}</p>
        <p>{output || operation || next ? next : '0'}</p>
      </div>
    );
  }
}

View.propTypes = {
  output: PropTypes.string,
  operation: PropTypes.string,
  next: PropTypes.string,
};

View.defaultProps = {
  output: '',
  operation: '',
  next: '',
};

export default View;
