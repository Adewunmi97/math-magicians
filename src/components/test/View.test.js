import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import View from '../View';

describe('test View functionality', () => {
  it('should render 6x2 correctly on the screen', () => {
    const { container } = render(<View output="6" operation="x" next="2" />);
    const viewScreen = screen.getByTestId('screen');
    expect(viewScreen).toHaveTextContent('6x2');
    expect(container).toMatchSnapshot();
  });
});
