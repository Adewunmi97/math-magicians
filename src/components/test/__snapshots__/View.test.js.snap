// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test View functionality should render 6x2 correctly on the screen 1`] = `
<div>
  <div
    class="View"
    data-testid="screen"
  >
    <p>
      6
    </p>
    <p>
      x
    </p>
    <p>
      2
    </p>
  </div>
</div>
`;
